modules/related-posts/jetpack-related-posts.php

```diff
@@ -1055,25 +1065,31 @@ protected function _get_related_post_ids( $post_id, $size, array $filters ) {
 			}
 		}
 
-		// Copy all valid cache values
-		$new_cache = array();
-		foreach ( $cache as $k => $v ) {
-			if ( is_array( $v ) && $v[ 'expires' ] > $now_ts ) {
-				$new_cache[ $k ] = $v;
+		$cache_ttl = 12 * HOUR_IN_SECONDS;
+
+		// Update cache
+		if ( wp_using_ext_object_cache() ) {
+			set_transient( $transient_name, $related_posts, $cache_ttl );
```

Comment:
`$related_posts` which is the normalized data from the curl request which could be a successful response but just without any data (empty) because there is currently nothing returned by the related post endpoint.

We've gotten bug reports before where posts gets delayed in syncing up to wpcom servers which results in empty related post responses and the client site caching the empty results for 12 hours. We should either not cache empty arrays or cache them for a shorter amount of time, perhaps 1 minute or something instead?